File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/dto/student/CreateStudentRequest.java
package org.example.olympic_ot_project.dto.student;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class CreateStudentRequest {
    @NotBlank(message = "Tên đăng nhập không được để trống")
    @Size(min = 10, max = 100, message = "Tên đăng nhập phải từ 10 ký tự trở lên")
    private String username;
    @NotBlank(message = "Mật khẩu không được để trống")
    @Size(min = 6, max = 100, message = "Mật khẩu phải từ 6 ký tự trở lên")
    private String password;
    @NotBlank(message = "Email không được để trống")
    @Email(message = "Email không đúng định dạng")
    private String email;
    @NotBlank(message = "Họ và tên không được để trống")
    @Size(min = 8 , message = "Họ và tên phải từ 8 ký tự đổ lên")
    private String fullName;
    @NotNull(message = "Lớp không được để trống")
    private Integer classId;
}

File: frontend/olympic_it_project/olympic_it_project/lib/dto/admin_manager/student/create_student_request.dart
class CreateStudentRequest {
  final String username;
  final String password;
  final String email;
  final String fullName;
  final int classId;

  CreateStudentRequest({
    required this.username,
    required this.password,
    required this.email,
    required this.fullName,
    required this.classId,
  });

  Map<String, dynamic> toJson() {
    return {
      "username": username,
      "password": password,
      "email": email,
      "fullName": fullName,
      "classId": classId,
    };
  }
}